pkg/kubelet/kubelet.go

```diff
@@ -2112,6 +2132,25 @@ func (kl *Kubelet) syncLoopIteration(updates <-chan kubetypes.PodUpdate, handler
 			// TODO: Do we want to support this?
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
+	case e := <-plegCh:
+		// Filter out started events since we don't use them now.
+		if e.Type == pleg.ContainerStarted {
+			break
+		}
+		pod, ok := kl.podManager.GetPodByUID(e.ID)
+		if !ok {
+			// If the pod no longer exists, ignore the event.
+			glog.V(4).Infof("SyncLoop (PLEG): ignore irrelevant event: %#v", e)
+			break
+		}
+		glog.V(2).Infof("SyncLoop (PLEG): %q, event: %#v", kubeletutil.FormatPodName(pod), e)
```

Comment:
SGTM.

On Thu, Nov 12, 2015 at 2:08 PM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/kubelet/kubelet.go
> <https://github.com/kubernetes/kubernetes/pull/13571#discussion_r44722224>
> :
>
> > @@ -2112,6 +2132,25 @@ func (kl *Kubelet) syncLoopIteration(updates <-chan kubetypes.PodUpdate, handler
> >  			// TODO: Do we want to support this?
> >  			glog.Errorf("Kubelet does not support snapshot update")
> >  		}
> > +	case e := <-plegCh:
> > +		// Filter out started events since we don't use them now.
> > +		if e.Type == pleg.ContainerStarted {
> > +			break
> > +		}
> > +		pod, ok := kl.podManager.GetPodByUID(e.ID)
> > +		if !ok {
> > +			// If the pod no longer exists, ignore the event.
> > +			glog.V(4).Infof("SyncLoop (PLEG): ignore irrelevant event: %#v", e)
> > +			break
> > +		}
> > +		glog.V(2).Infof("SyncLoop (PLEG): %q, event: %#v", kubeletutil.FormatPodName(pod), e)
>
> This would only happen if there is a container event, which would be
> useful to know IMO. I think it would be okay as V(3) though. How about I do
> a followup PR to change logging levels of all messages in this function?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13571/files#r44722224>.
>